ipld_dag_pb/node.py

```diff
@@ -71,6 +71,53 @@ class RawPBLink:
     hash: BytesLike
 
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, RawPBLink):
+            return NotImplemented
+
+        if (getattr(self, "name", None) is not None) and getattr(self, "name", None):
+            if self.name != other.name:
+                return False
```

Comment:
I don't understand what the first conditional is doing? Shouldn't this just be:

```suggestion
        if getattr(self, "name", None) != getattr(other, "name", None):
            return False
```

Reply:
yes, your suggestion is right.

the intended conditional structure is:
```python
        if (getattr(self, "name", None) is not None) and (getattr(self, "name", None) is not None):
            if self.name != other.name:
                return False
```

which i think works as well but reads less easily than your suggestion, so I'd include changes to this in the next commit as well.
i basically reused the same line as above in the remaining conditions and erroneously left out the `is not None` part of the second conditions, so I'd correct those as well.